Yoast/javascript
# Remove functionality used for building keyphrase forms and German form creation

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Removes obsolete functionality to build keyphrase forms and German word forms.

## Relevant technical choices:

* Removes functionality for:
  * building keyword forms (language-independent)
  * building German word forms
* Refactors the following functionality:
  * The helper function `getLanguagesWithWordFormSupport` checks whether a given language has morphology support. This check used to be based on whether there was a `getForms` function for a given language. As we don't have `getForms` functions anymore, this has been changed so that now it's based on whether there is a `determineStem` function for that language.
* Please note that some of the changes in the structure went hand in hand with changes in the code. This means that a morphology data release will be necessary. If we want to avoid having to do a morphology data release, this would be possible if some of the changes are reversed. So we need to decide whether we want to prioritize a cleaner data file, or whether we prioritize being able to avoid a data release.

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test` and make sure that all specs still pass.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-167
